plugins/woocommerce/includes/wc-core-functions.php

```diff
@@ -2512,19 +2512,15 @@ function wc_selected( $value, $options ) {
 function wc_get_server_database_version() {
 	global $wpdb;
 
-	if ( empty( $wpdb->is_mysql ) ) {
+	if ( empty( $wpdb->is_mysql ) || ! $wpdb->use_mysqli ) {
 		return array(
 			'string' => '',
 			'number' => '',
 		);
 	}
 
 	// phpcs:disable WordPress.DB.RestrictedFunctions, PHPCompatibility.Extensions.RemovedExtensions.mysql_DeprecatedRemoved
-	if ( $wpdb->use_mysqli ) {
-		$server_info = mysqli_get_server_info( $wpdb->dbh );
-	} else {
-		$server_info = mysql_get_server_info( $wpdb->dbh );
-	}
+	$server_info = mysqli_get_server_info( $wpdb->dbh );
 	// phpcs:enable WordPress.DB.RestrictedFunctions, PHPCompatibility.Extensions.RemovedExtensions.mysql_DeprecatedRemoved
```

Comment:
(Optional thing) given this change, I think we could do some further tidying up:

```php
	// phpcs:ignore WordPress.DB.RestrictedFunctions
	$server_info = mysqli_get_server_info( $wpdb->dbh );
```

(Since we've reduced things to a single statement, we only need a single `phpcs:ignore`—instead of the paired disable/enable comments—and `mysql_DeprecatedRemoved` is no longer relevant.)

Reply:
Good catch, one sec...

Reply:
Fixed in [2019900](https://github.com/woocommerce/woocommerce/pull/34336/commits/20199007e305a5ae089b9c354ec811a85d4190f2).